src/lava/lib/optimization/solvers/generic/models.py

```diff
@@ -0,0 +1,318 @@
+    def __init__(self, proc):
+        # Instantiate child processes
+        # The input shape is a 2D vector (shape of the weight matrix).
+        wta_weight = -2
+        shape = proc.proc_params.get("shape", (1,))
+        diagonal = proc.proc_params.get("cost_diagonal")
+        weights = proc.proc_params.get(
+            "weights",
+            wta_weight
+            * np.logical_not(np.eye(shape[1] if len(shape) == 2 else 0)),
+        )
+        noise_amplitude = proc.proc_params.get("noise_amplitude", 1)
+        steps_to_fire = proc.proc_params.get("steps_to_fire", 10)
+        importances = proc.proc_params.get("importances", 10)
+        self.s_bit = StochasticIntegrateAndFire(shape=shape,
+                                                increment=importances,
+                                                noise_amplitude=noise_amplitude,
+                                                steps_to_fire=steps_to_fire,
+                                                cost_diagonal=diagonal)
+        if weights.shape != (0, 0):
+            self.dense = Dense(weights=weights)
+            self.s_bit.out_ports.messages.connect(self.dense.in_ports.s_in)
+            self.dense.out_ports.a_out.connect(self.s_bit.in_ports.added_input)
+
+        # Connect the parent InPort to the InPort of the Dense child-Process.
+        proc.in_ports.a_in.connect(self.s_bit.in_ports.added_input)
+        # Connect the OutPort of the LIF child-Process to the OutPort of the
+        # parent Process.
+        self.s_bit.out_ports.messages.connect(proc.out_ports.s_out)
+        self.s_bit.out_ports.local_cost.connect(
+            proc.out_ports.local_cost
+        )
+        proc.vars.variable_assignment.alias(self.s_bit.prev_assignment)
+
+
+@implements(proc=CostConvergenceChecker, protocol=LoihiProtocol)
+@requires(CPU)
+class CostConvergenceCheckerModel(AbstractSubProcessModel):
+    def __init__(self, proc):
+        # Instantiate child processes
+        # The input shape is a 2D vector (shape of the weight matrix).
+        shape = proc.proc_params.get("shape", (1,))
+        weights = proc.proc_params.get("weights", np.ones((1, shape[0])))
+        self.dense = Dense(weights=weights, num_message_bits=24)
+        self.cost_integrator = CostIntegrator(shape=(1,))
+        self.dense.out_ports.a_out.connect(
+            self.cost_integrator.in_ports.cost_in
+        )
+
+        # Connect the parent InPort to the InPort of the Dense child-Process.
+        proc.in_ports.cost_components.connect(self.dense.in_ports.s_in)
+
+        # Connect the OutPort of the LIF child-Process to the OutPort of the
+        # parent Process.
+        self.cost_integrator.out_ports.update_buffer.connect(
+            proc.out_ports.update_buffer)
+        proc.vars.min_cost.alias(self.cost_integrator.vars.min_cost)
+
+
+@implements(proc=CostIntegrator, protocol=LoihiProtocol)
+@requires(CPU)
+class CostIntegratorModel(PyLoihiProcessModel):

```

Comment:
Why is there a CostIntegrator Process / Model here? There is already one in the public Lava.